### DISCUSSION TITLE
NVIDIA L40 Compatibility

### DISCUSSION THREAD

**Discussion by marshp7516** [AdaptiveCpp/AdaptiveCpp#1517] [General]
Hello,

I created a simple matrix addition program utilizing SYCL. I compiled it successfully with the ACPP compiler, but when I try to run it on an NVIDIA L40 GPU, I encounter the following error message:
![image](https://github.com/user-attachments/assets/6d20a988-e7b0-43ff-88f4-f2dd95550850)

I was wondering if support for the Ada series GPUs would be added in the future. I assume that the unsupported GPU is causing these errors, as it seems no supported GPU is being detected by the queue, but it would be nice if someone could confirm it:
![image](https://github.com/user-attachments/assets/dbb0b244-6f1b-41d7-8e22-632193336ee8)

Thank you for any help!

**Comment by illuhad** [comment]
You are seeing a JIT failure because LLVM does not recognize the GPU architecture. Which GPU architectures are supported and recognized is not down to AdaptiveCpp, but depends on the LLVM version that you have built AdaptiveCpp against.

Which LLVM version are you using? Recent LLVM versions should have support for `sm_89`.

**Reply by fodinabor** [reply]
(For reference, `sm_89` was added in LLVM 16: https://github.com/llvm/llvm-project/commit/9a01cca66036087e4da37c221a4b911818910524)

**Reply by illuhad** [reply]
Thanks, was also just trying to find out which version introduced support, but you beat me to it ;)

**Reply by marshp7516** [reply]
Thank you everyone! This worked for me.
